transformer_engine/pytorch/attention.py

```diff
@@ -1439,10 +1546,27 @@ def forward(
         if softmax_scale is None:
             softmax_scale = q.shape[-1] ** (-0.5)
 
+        if isinstance(cp_group, list):
```

Comment:
Is it robust to rely on whether `cp_group` is a `list` here for determination of `a2a+p2p`? What if it's a list of one integer: `[2]`.

Could you also add in the comment or docstring somewhere that `cp_group[0]` is the A2A group, and `cp_group[1]` is the P2P group, so that it's more readable for users/developers. I had to remind myself sometimes.

Reply:
I think it's robust, because otherwise `get_distributed_world_size` and `get_distributed_rank` would fail. I also have some check [at here](https://github.com/xrennvidia/TransformerEngine/blob/xren/cp_a2a_p2p/transformer_engine/pytorch/attention.py#L4091-L4105).

[I added more explanations of cp_group in doc_string.](https://github.com/NVIDIA/TransformerEngine/pull/1209/commits/582a4e03de694b75869845bf95923fb6a252cfde).